tf-modules/archive/process_dead_letter_archive.tf

```diff
@@ -0,0 +1,116 @@
+data "aws_iam_policy_document" "process_dead_letter_archive_policy" {
+  statement {
+    actions = ["dynamodb:UpdateItem"]
+    resources = [
+      var.dynamo_tables.executions.arn,
+      var.dynamo_tables.granules.arn,
+      var.dynamo_tables.pdrs.arn
+    ]
+  }
+
+  statement {
+    actions = [
+      "states:DescribeExecution",
+      "states:GetExecutionHistory"
+    ]
+    resources = ["*"]
+  }
+
+  statement {
+    actions   = [
+      "s3:GetObject",
+      "s3:PutObject",
+      "s3:DeleteObject"
+    ]
+    resources = ["arn:aws:s3:::${var.system_bucket}/*"]
+  }
+
+  statement {
+    actions   = [
+      "s3:ListBucket"
+    ]
+    resources = ["arn:aws:s3:::${var.system_bucket}"]
+  }
+
+  statement {
+    actions = [
+      "ec2:CreateNetworkInterface",
+      "ec2:DescribeNetworkInterfaces",
+      "ec2:DeleteNetworkInterface"
+    ]
+    resources = ["*"]
+  }
+
+  statement {
+    actions = [
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:DescribeLogStreams",
+      "logs:PutLogEvents"
+    ]
+    resources = ["*"]
+  }
+
+  statement {
+    actions = [
+      "s3:GetObject*",
+    ]
+    resources = [for b in local.allowed_buckets: "arn:aws:s3:::${b}/*"]
+  }
+
+  statement {
+    actions = [
+      "secretsmanager:GetSecretValue"
+    ]
+    resources = [var.rds_user_access_secret_arn]
+  }
+}
+
+resource "aws_iam_role" "process_dead_letter_archive_role" {
+  name                 = "${var.prefix}_process_dead_letter_archive_role"
+  assume_role_policy   = data.aws_iam_policy_document.lambda_assume_role_policy.json
+  permissions_boundary = var.permissions_boundary_arn
+  tags                 = var.tags
+}
+
+resource "aws_iam_role_policy" "process_dead_letter_archive_role_policy" {
+  name   = "${var.prefix}_process_dead_letter_archive_lambda_role_policy"
+  role   = aws_iam_role.process_dead_letter_archive_role.id
+  policy = data.aws_iam_policy_document.process_dead_letter_archive_policy.json
+}
+
+resource "aws_lambda_function" "process_dead_letter_archive" {
+  filename         = "${path.module}/../../packages/api/dist/processDeadLetterArchive/lambda.zip"
+  source_code_hash = filebase64sha256("${path.module}/../../packages/api/dist/processDeadLetterArchive/lambda.zip")
+  function_name    = "${var.prefix}-processDeadLetterArchive"
+  role             = aws_iam_role.process_dead_letter_archive_role.arn
+  handler          = "index.handler"
+  runtime          = "nodejs12.x"
+  timeout          = 300
+  memory_size      = 512
+
+  environment {
+    variables = {
+      system_bucket   = var.system_bucket
+      stackName       = var.prefix
+      ExecutionsTable = var.dynamo_tables.executions.name
+      GranulesTable   = var.dynamo_tables.granules.name
+      PdrsTable       = var.dynamo_tables.pdrs.name
+      databaseCredentialSecretArn = var.rds_user_access_secret_arn
+      RDS_DEPLOYMENT_CUMULUS_VERSION = "5.0.0"
```

Comment:
Should this lambda also take the RDS heartbeat env var?    For that matter, `write_db_dlq_records_to_s3` and even sf_event_sqs_to_db_records?       🤔

Reply:
I'm not sure!

Reply:
I think it should - we're using the heartbeat config 'globally' in the cumulus model - it causes the db model to run a verification query on a retry, which reduces performance but increases reliability in serverless situations where the db *might* be down.

Reply:
https://github.com/nasa/cumulus/pull/2181/commits/1a1b6190970b18545ddac5359fb8bbb192c664a1

Reply:
For the record, `write_db_dlq_records_to_s3` does not interact with the DB, only the SQS DLQ and S3.